cmd/kaspawallet/sweep.go

```diff
@@ -0,0 +1,267 @@
+package main
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/kaspanet/go-secp256k1"
+	"github.com/kaspanet/kaspad/cmd/kaspawallet/daemon/client"
+	"github.com/kaspanet/kaspad/cmd/kaspawallet/daemon/pb"
+	"github.com/kaspanet/kaspad/cmd/kaspawallet/libkaspawallet"
+	"github.com/kaspanet/kaspad/cmd/kaspawallet/libkaspawallet/serialization"
+	"github.com/kaspanet/kaspad/cmd/kaspawallet/utils"
+	"github.com/kaspanet/kaspad/domain/consensus/model/externalapi"
+	"github.com/kaspanet/kaspad/domain/consensus/utils/consensushashing"
+	"github.com/kaspanet/kaspad/domain/consensus/utils/constants"
+	"github.com/kaspanet/kaspad/domain/consensus/utils/subnetworks"
+	"github.com/kaspanet/kaspad/domain/consensus/utils/txscript"
+	"github.com/kaspanet/kaspad/domain/consensus/utils/utxo"
+	"github.com/kaspanet/kaspad/domain/dagconfig"
+	"github.com/kaspanet/kaspad/domain/miningmanager/mempool"
+	"github.com/kaspanet/kaspad/util"
+	"github.com/kaspanet/kaspad/util/txmass"
+	"github.com/pkg/errors"
+)
+
+const feePerInput = 10000
+
+func sweep(conf *sweepConfig) error {
+
+	//Sweep assumes the passed private key is a schnorr private key: I see no function to explictly test the type of key
+	//It will almost certainly fail somewhere if this is not the case
+	privateKeyBytes, err := hex.DecodeString(conf.PrivateKey)
+	if err != nil {
+		return err
+	}
+
+	if err != nil {
+		return err
+	}
+	publicKeybytes, err := libkaspawallet.PublicKeyFromPrivateKey(privateKeyBytes)
+	if err != nil {
+		return err
+	}
+
+	//NewAddress might seem confusing, but the function is entirely deterministic based on the public key
```

Comment:
Please delete this comment. If you think the name is misleading you can change it from `NewAddressPublicKey` to `AddressPublicKeyFromPublicKey` or something like this (but then you should also change `NewAddressPublicKeyECDSA` etc to keep the same convention)